### DISCUSSION TITLE
Running systemd with `--runtime io.containerd.kata.v2`

### DISCUSSION THREAD

**Discussion by aucampia** [kata-containers/kata-containers#7357] [Q&A]
I want to run systemd inside a container with the kata runtime, I tried:

```bash
docker run --rm --runtime io.containerd.kata.v2 -it docker.io/nestybox/ubuntu-jammy-systemd
```

When I run this I get exit status 255. The systemd-journal, `docker events` and `kata-runtime env` output for this can be seen [here](https://gist.github.com/aucampia/235d8c178b0b3c0fb9fdf6af5f1acf15).

Are there any examples somewhere of how to make systemd run inside a kata runtime?

I suspect the problem may be related to

```
Jul 17 16:37:13 ... kernel: eth0: Caught tx_queue_len zero misconfig
```

## Edit #1

It can't be `kernel: eth0: Caught tx_queue_len zero misconfig` because this shows up when this runs succesfully:

```console
$ docker run  --runtime io.containerd.kata.v2 --rm docker.io/library/alpine sh -c 'cat /proc/cmdline; uname -a'; echo $?
tsc=reliable no_timer_check rcupdate.rcu_expedited=1 i8042.direct=1 i8042.dumbkbd=1 i8042.nopnp=1 i8042.noaux=1 noreplace-smp reboot=k cryptomgr.notests net.ifnames=0 pci=lastbus=0 root=/dev/pmem0p1 rootflags=dax,data=ordered,errors=remount-ro ro rootfstype=ext4 console=hvc0 console=hvc1 quiet systemd.show_status=false panic=1 nr_cpus=16 selinux=0 systemd.unit=kata-containers.target systemd.mask=systemd-networkd.service systemd.mask=systemd-networkd.socket scsi_mod.scan=none
Linux d636d340d7f9 5.19.2 #2 SMP Tue Jul 4 06:39:27 UTC 2023 x86_64 Linux
0
```

**Comment by aggroot** [comment]
Hey @aucampia, I'm trying to achieve the same thing. Did you find any solution to this?
